### DISCUSSION TITLE
Chapter9 - training strategy - coding practice problem

### DISCUSSION THREAD

**Discussion by andreicorni25** [Mayo-Radiology-Informatics-Lab/MIDeL#11] [Q & A] [answered]
Hello I am trying to solve chapter 9 training practic, and when I try to evaluate the modelfor the given image I get this error:

Can someone explain to me why do I get this error? 

Thank you!

Here is my code:
`from monai.transforms import AddChannel
from skimage.io import imread
import numpy as np
import cv2
from torch.utils.data import DataLoader
from torchvision import models

img_array = imread(train_imageinfo_list[0][0])
resized_img = cv2.resize(img_array, (224, 224))
img_tensor = torch.from_numpy(resized_img)

channel_adder = AddChannel()
channel_image = channel_adder(img_tensor)
batch_image = channel_adder(channel_image)
img_tensor = batch_image

model= models.vgg16()
model(img_tensor)`

**Comment by BardiaKh** [comment, marked as answer]
To use torchvision models, your input should have 3 channels. You can use MONAI's `RepeatChannel` transform to achieve this. For example:

```python
from monai.transforms import RepeatChannel

channel_repeater = RepeatChannel(3)
img = channel_repeater(img)
```

[Here is the link](https://docs.monai.io/en/stable/transforms.html#repeatchannel) to complete documentation of this transform.
